snappy/click.go

```diff
@@ -44,27 +44,39 @@ import (
 var killWait = 5 * time.Second
 
 // servicesBinariesStringsWhitelist is the whitelist of legal chars
-// in the "binaries" and "services" section of the package.yaml
+// in the "binaries" and "services" section of the snap.yaml
 var servicesBinariesStringsWhitelist = regexp.MustCompile(`^[A-Za-z0-9/. _#:-]*$`)
 
 // generate the name
-func generateBinaryName(m *packageYaml, binary Binary) string {
+func generateBinaryName(m *snapYaml, app *AppYaml) string {
 	var binName string
 	if m.Type == snap.TypeFramework {
-		binName = filepath.Base(binary.Name)
+		binName = filepath.Base(app.Name)
 	} else {
-		binName = fmt.Sprintf("%s.%s", m.Name, filepath.Base(binary.Name))
+		binName = fmt.Sprintf("%s.%s", m.Name, filepath.Base(app.Name))
 	}
 
 	return filepath.Join(dirs.SnapBinariesDir, binName)
 }
 
-func binPathForBinary(pkgPath string, binary Binary) string {
-	return filepath.Join(pkgPath, binary.Exec)
+func binPathForBinary(pkgPath string, app *AppYaml) string {
+	return filepath.Join(pkgPath, app.Command)
 }
 
-func verifyBinariesYaml(binary Binary) error {
-	return verifyStructStringsAgainstWhitelist(binary, servicesBinariesStringsWhitelist)
+func verifyAppYaml(app *AppYaml) error {
+	return verifyStructStringsAgainstWhitelist(*app, servicesBinariesStringsWhitelist)
+}
+
+func verifyUsesYaml(uses *usesYaml) error {
+	if err := verifyStructStringsAgainstWhitelist(*uses, servicesBinariesStringsWhitelist); err != nil {
+		return err
+	}
+
+	if uses.Type != "migration-skill" {
+		return fmt.Errorf("can not use skill %q, only migration-skill supported", uses.Type)
+	}
```

Comment:
There's no need to. The migration-skill is an aberration we're putting in place to move people forward comfortably. We don't need a complete list here.